src/StatementDumper.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Helmich\TypoScriptParser;
+
+use Helmich\TypoScriptParser\Parser\AST\ConditionalStatementTerminator;
+use Helmich\TypoScriptParser\Parser\AST\NopStatement;
+use Helmich\TypoScriptParser\Parser\AST\Statement;
+
+class StatementDumper
+{
+    private bool $dumpLineNumbers;
+
+    /**
+     * @param array{dumpLineNumbers?: bool} $options
+     */
+    public function __construct(array $options = [])
+    {
+        $this->dumpLineNumbers = $options['dumpLineNumbers'] ?? false;
+    }
+
+    /**
+     * @param array<Statement> $statements
+     */
+    public function dump(array $statements): string
+    {
+        return $this->dumpRecursive($statements);
+    }
+
+    private function dumpRecursive(mixed $statement): string
```

Comment:
Albeit unwieldy, but we could specify a more specific type hint instead of `mixed`:

```suggestion
    /**
     * @param Node|Node[]|ConditionalStatementTerminator|null $statement
     */
    private function dumpRecursive(Node|array|ConditionalStatementTerminator|null $statement): string
```

Reply:
after applying the suggested changes, i get a lot more problems with phpstan :cry:

Reply:
Yeah, I saw. 😓  In any case, I think this is solved very cleanly know, even with the remaining `mixed` type hint. :+1: